src/Functions/URL/resolveRelativeURL.cpp

```diff
@@ -0,0 +1,218 @@
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/FunctionFactory.h>
+#include <base/find_symbols.h>
+#include <Functions/URL/protocol.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+extern const int ILLEGAL_COLUMN;
+}
+
+bool isAbsoluteURL(const char * url_begin, const char * url_end)
+{
+    Pos pos = find_first_symbols<':'>(url_begin, url_end);
+    if (pos == url_end)
+    {
+        return false;
+    }
+    return pos[1] == '/';
+}
+
+Pos getDomainNameBeginPos(const char * url_begin, const char * url_end)
+{
+    Pos pos = find_first_symbols<':'>(url_begin, url_end);
+    if (pos == url_end)
+    {
+        return pos;
+    }
+    bool has_subsequent_slash = pos[1] == '/' && pos[2] == '/';
```

Comment:
Same, please check that if works on corner-cases like empty and one-char strings